src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -134,17 +142,53 @@ public override void Initialize (AnalysisContext context)
 					CheckCalledMember (operationContext, methodSymbol, dangerousPatterns);
 				}, OperationKind.DelegateCreation);
 
+				static void CheckStaticConstructors (OperationAnalysisContext operationContext,
```

Comment:
We could abstract this method away and share it between the analyzers.

Reply:
I'm testing abstracting to a base class the common code in the analyzers, but I was expecting to open a different PR for that